### PR TITLE
Update categoryLocation model

### LINKED ISSUE #61: Add categoryLocation tests
We need to have tests for the categoryLocation model.

### PR DESCRIPTION
**Pull request by Ofekmn**
adds verification and the ability to update.

resolves #61

### CODE REVIEW

**Review by Sendy459 — DISMISSED** [review]

File: category_location/tests.py [hunk 1]
```diff
@@ -1,3 +1,41 @@
...
+class TestCategoryModel:
+
+    def test_create_category(self, category1):
+        category1.save()
+        assert category1 in Category.objects.all()
+
+    def test_delete_category_location(self, category_location1):
+        category_location1.delete()
+        with pytest.raises(CategoryLocation.DoesNotExist):
+            CategoryLocation.objects.get(pk=category_location1.pk)
```
Comment (talko611) [inline category_location/tests.py:41]: There is a missing assert check
can also check this in this way:
`category_location1.delete()`
`assert category_location1 not in list(CategotryLocation.objects.all())`

**Review by talko611 — COMMENTED** [review]
I added comment
Also, you should check update methods

File: category_location/tests.py (hunk 1 above)
Comment (Ofekmn) [reply category_location/tests.py:41]: There is no need to add an assert check.
Pytest expects the CategoryLocation.DoesNotExist exception, and will fail the test if it isn't raised.

**Comment by ormergi** [comment]
Hi Ofek, please take a look at my comment from another PR about tests that doesn't test the app business logic https://github.com/redhat-beyond/FitMeet/pull/26#discussion_r1171429021

**Review by rotem2022 — APPROVED** [review]
Looks Good

**Review by talko611 — APPROVED** [review]

**Review by SaarOrus — DISMISSED** [review]

**Review by rotem2022 — DISMISSED** [review]

**Review by Sendy459 — DISMISSED** [review]

**Review by talko611 — APPROVED** [review]

File: category_location/tests.py [hunk 2]
```diff
@@ -1,3 +1,66 @@
...
+STREET_NUMBER = 1
+INDOOR = False
+DESCRIPTION = "description1"
+
+OTHER_CATEGORY_NAME = 'category2'
+OTHER_LOCATION_NAME = 'location2'
+OTHER_CITY = "city2"
+OTHER_STREET = "street2"
+OTHER_INDOOR = False
+OTHER_DESCRIPTION = "description2"
```
Comment (ygalblum) [inline category_location/tests.py:20]: You're not using any of these constants in the code. Did you mean to use them in `test_update_category_location`?

**Review by ygalblum — CHANGES_REQUESTED** [review]

**Review by Sendy459 — DISMISSED** [review]

File: category_location/tests.py [hunk 3]
```diff
@@ -1,3 +1,69 @@
...
+        assert category_location1.category == category1
+        assert category_location1.location == location1
+        with pytest.raises(IntegrityError):
+            CategoryLocation.objects.create(category=category1, location=location1)
+
+    def test_update_category_location(self,
+                                      category_location1, category1, location1):
+        assert category_location1.category == category1
+        assert category_location1.location == location1
+        category_location1.update(category=category1, location=location1)
```
Comment (ygalblum) [inline category_location/tests.py:61]: Why do you need this line?

Comment (Ofekmn) [reply category_location/tests.py:61]: When I did the PR that had Categoryn tests, Kobi suggested that I should check that the fields are correct before the update(https://github.com/redhat-beyond/FitMeet/pull/58)
I figured it should be like that here as well.

**Review by rotem2022 — APPROVED** [review]

File: category_location/tests.py (hunk 3 above)
Comment (ygalblum) [reply category_location/tests.py:61]: The `assert` lines I understand. But, then you update to the same values and don't check that the update happened (not sure you should). Am I reading this wrong? If I'm right, I don't understand why it is needed. If I'm wrong, can you please explain?

**Review by SaarOrus — APPROVED** [review]

**Review by ygalblum — APPROVED** [review]

**Pull request merged**